src/flask_pyoidc/flask_pyoidc.py

```diff
@@ -88,42 +88,29 @@ def init_app(self, app):
             for (name, configuration) in self._provider_configurations.items()
         }
 
-    def _get_post_logout_redirect_uri(self, client):
-        if client.post_logout_redirect_uris:
-            return client.post_logout_redirect_uris[0]
-        return self._get_url_for_logout_view()
-
-    def _get_url_for_logout_view(self):
-        if not self._logout_view:
-            return None
-
+    def _get_urls_for_logout_views(self):
         try:
-            return url_for(self._logout_view.__name__, _external=True)
+            return [self._get_url_for_view_function(view) for view in self._logout_views]
         except BuildError:
             logger.error('could not build url for logout view, it might be mounted under a custom endpoint')
             raise
 
-    def _register_client(self, client):
-        def default_post_logout_redirect_uris():
-            url_for_logout_view = self._get_url_for_logout_view()
-            if url_for_logout_view:
-                return [url_for_logout_view]
-            return []
+    def _get_url_for_view_function(self, view_func):
+        for endpoint, f in current_app.view_functions.items():
+            if f == view_func:
```

Comment:
This if condition will never be true if you are using flask extensions that wrap view functions to generate OpenAPI documentation (Swagger). For example if you use `flask-restx`, the view function object becomes `<function Logout.get at 0x000001DF5DB27760>` and is registered with flask app as:

```python
>>> app.view_functions
{'apiv2.logout_logout': <function View.as_view.<locals>.view at 0x000001DF5DB40670>
```

In an enterprise environment, OpenAPI compliance is a standard so all view functions are likely to be wrapped with extensions that automatically generate swagger documentation for APIs. We must not have conflicts with the use of other flask-extensions.

Reply:
That is due to the ordering of the decorators: `@auth.oidc_logout` needs to be the second outermost decorator, right after the Flask `@app.route(...)`:
```python
@app.route('/logout')
@auth.oidc_logout
@access_log
@....
def logout():
    return "You've been successfully logged out!"
```

Reply:
I did it in #124. I'm using `flask.request.url` instead.

Reply:
I think it would be good if we aligned on a single PR to not duplicate the efforts 🙂 .

I checked #124 and it has some unrelated changes now (e.g. the imports), and this one is ready to go, so I'd like to merge this one.
Is there anything more that should be added to this one from #124?

Reply:
Our approach is exactly same 😂. I have left some reviews to address the changes.

Reply:
Hehe! :)

Thanks a ton for the detailed review, I've addressed the comments! 🙇